# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: image/DecodePool.cpp
# Commit: 1d9384906287
# Full Hash: 1d93849062870a8d492eeef2150ecea174cb6223
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-26 21:12:53
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r f5ebdcca640d -r 1d9384906287 image/DecodePool.cpp
--- a/image/DecodePool.cpp	Sun May 26 14:29:56 2019 +0000
+++ b/image/DecodePool.cpp	Sun May 26 14:30:14 2019 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/ClearOnShutdown.h"
 #include "mozilla/DebugOnly.h"
 #include "mozilla/Monitor.h"
+#include "mozilla/StaticPrefs.h"
 #include "mozilla/TimeStamp.h"
 #include "nsCOMPtr.h"
 #include "nsIObserverService.h"
@@ -20,8 +21,6 @@
 #include "prsystem.h"
 #include "nsIXULRuntime.h"
 
-#include "gfxPrefs.h"
-
 #include "Decoder.h"
 #include "IDecodingTask.h"
 #include "RasterImage.h"
@@ -354,7 +353,7 @@
 
 DecodePool::DecodePool() : mMutex("image::IOThread") {
   // Determine the number of threads we want.
-  int32_t prefLimit = gfxPrefs::ImageMTDecodingLimit();
+  int32_t prefLimit = StaticPrefs::ImageMTDecodingLimit();
   uint32_t limit;
   if (prefLimit <= 0) {
     int32_t numCores = NumberOfCores();
@@ -384,7 +383,7 @@
   uint32_t idleLimit;
 
   // The timeout period before shutting down idle threads.
-  int32_t prefIdleTimeout = gfxPrefs::ImageMTDecodingIdleTimeout();
+  int32_t prefIdleTimeout = StaticPrefs::ImageMTDecodingIdleTimeout();
   TimeDuration idleTimeout;
   if (prefIdleTimeout <= 0) {
     idleTimeout = TimeDuration::Forever();